### PR TITLE
227 pp new dataset notification

### PR DESCRIPTION
**Pull request by papesci**
implementation of a check for new or deleted dataset.
a new option has been added to cate cli.
cate ds list -u 
show the list of dataset and at the end report for new or deleted datatset.

### CODE REVIEW

**Comment by forman** [comment]
Hi @papesci, to which issue does this PR relate to?

Please always update `CHANGES.md` so our testers and users can see what has changed!

File: cate/core/ds.py [hunk 1]
```diff
@@ -338,6 +338,18 @@ def invalidate(self):
...
 
+    def get_updates(self, reset=False) -> Dict:
+        """
+        Ask the datastore to retrieve the differences found between a previous
+        dataStore status and the current one,
+        The implementation should return a dictionary with the new ['new'] and removed ['del'] dataset.
+        it also return the reference time to the datastore status taken as previous, it could
+        be useful to know to which time the found differences are compared to.
+        Reset flag is used to clean up the support files, freeze and diff.
+        :return:
```
Comment (forman) [inline cate/core/ds.py]: Add docs for `:return:`

File: cate/core/ds.py [hunk 2]
```diff
@@ -338,6 +338,18 @@ def invalidate(self):
         """
         pass
 
+    def get_updates(self, reset=False) -> Dict:
+        """
+        Ask the datastore to retrieve the differences found between a previous
+        dataStore status and the current one,
+        The implementation should return a dictionary with the new ['new'] and removed ['del'] dataset.
+        it also return the reference time to the datastore status taken as previous, it could
+        be useful to know to which time the found differences are compared to.
```
Comment (forman) [inline cate/core/ds.py]: Avoid "could" in operation specification. Is the framework using the reference time or not? Be clear.

**Comment by papesci** [comment]
Hi @forman thanks for the review. Rerquested changes has been made and an additional test has beend added

**Pull request merged**